### PR TITLE
feat: partial support for TES-on-Azure

### PR DESCRIPTION
**Pull request by uniqueg**
- TES-on-Azure now works for the example where Snakemake sends a workflow to it directly (but not via the gateway; that will require fixes in proTES); TES-on-Azure basic auth credentials will need to be passed via `FUNNEL_SERVER_USER` and `FUNNEL_SERVER_PASSWORD` (but this will change in Snakemake 8+, where the behavior of the TES and other executors will be different)
- Clean up `README.md` and notebook; remove mention of FTP storage
- Update Conda environment file; among other issues, limit to Snakemake <8 (which will have a different way of using the TES executor
- Simplify directory structure and place workflow file in dedicated directory; this prevents the TES executor from attaching various files to each TES call that are not needed
- Remove `.inputs`

### CODE REVIEW

File: demo.ipynb [hunk 1]
```diff
@@ -98,40 +89,36 @@
...
-    "    )\n",
-    "    if [[ $TASK_ID == \"null\" ]]; then\n",
-    "        echo \"FAILED\"\n",
-    "    else\n",
-    "        echo \"Task ID: $TASK_ID\"\n",
-    "        TASKS[[ $TASK_ID ]]=\"$TES\"\n",
-    "    fi\n",
+    "    snakemake \\\n",
+    "        --directory wf-hashsplitter \\\n",
+    "        --jobs 1 \\\n",
```
Comment (vschnei) [inline demo.ipynb:104]: add  `--snakefile wf-hashsplitter/Snakefile \`

File: demo.ipynb [hunk 2]
```diff
@@ -142,53 +129,16 @@
...
-    "\n",
-    "\n",
-    "if [[ $TASK_ID == \"null\" ]]; then\n",
-    "    echo \"FAILED\"\n",
-    "else\n",
-    "    echo \"Task ID: $TASK_ID\"\n",
-    "fi\n",
+    "snakemake \\\n",
+    "    --directory wf-hashsplitter \\\n",
+    "    --jobs 1 \\\n",
```
Comment (vschnei) [inline demo.ipynb:136]: add  `--snakefile wf-hashsplitter/Snakefile \`

File: wf-hashsplitter/Snakefile [hunk 3]
```diff
@@ -5,16 +5,16 @@ localrules: all, download
 # any other remote storage is possible as well e.g. FTP
 # Remote storage has to be suported by Snakemake.
 # https://snakemake.readthedocs.io/en/stable/snakefiles/remote_files.html
-S3 = S3RemoteProvider(access_key_id="XXX", secret_access_key="XXX", endpoint_url="https://s3-elixir.cloud.e-infra.cz")
+S3 = S3RemoteProvider(access_key_id="8vVw4pzA4CLAg4VI", secret_access_key="yI2j5Y4IDEqWEGyewnmGUv4dXiGZy7OR", endpoint_url="https://s3-elixir.cloud.e-infra.cz")
```
Comment (vschnei) [inline wf-hashsplitter/Snakefile:8]: why did you add the credentials?

Comment (uniqueg) [reply wf-hashsplitter/Snakefile:8]: Oopsie. I'll fix it in a minute 🙈 One other reason why I don't like this approach by Snakemake, this is so easily forgotten

File: demo.ipynb (hunk 1 above)
Comment (uniqueg) [reply demo.ipynb:104]: It's not necessary if the file is called `Snakefile` and inside `--directory`. But I agree, why not be explicit. I'll add it together with the credentials
